gherkin/flagd-json-evaluator.feature

```diff
@@ -61,3 +63,12 @@ Feature: flagd json evaluation
       | "3.0.1" | "minor" |
       | "3.1.0" | "major" |
       | "4.0.0" | "none"  |
+
+  Scenario Outline: Errors and edge cases
+    When an integer flag with key <key> is evaluated with default value 13
+    Then the returned value should be <value>
+    Examples:
+      | key                                 | value |
+      | "error-targeting-flag"              | 2     |
+      | "missing-variant-targeting-flag"    | 2     |
+      | "non-string-variant-targeting-flag" | 2     |
```

Comment:
These all fail in the Java in-process provider, and pass in the Java rpc provider.

Reply:
yeah, the in-process providers fallback to default value if targeting rule evaluation fail.